pkgs/applications/virtualization/podman-desktop/default.nix

```
@@ -0,0 +1,119 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchYarnDeps
+, yarn
+, fixup_yarn_lock
+, nodejs
+, makeWrapper
+, copyDesktopItems
+, desktopToDarwinBundle
+, electron
+, makeDesktopItem
+}:
+
+stdenv.mkDerivation (self: {
+  pname = "podman-desktop";
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = "podman-desktop";
+    rev = "v${self.version}";
+    sha256 = "sha256-gEjcI+bfETYZB/pHDXRcNxNVDsbwuqQL1E22fMkIJHI=";
+  };
+
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${self.src}/yarn.lock";
+    sha256 = "sha256-x0hqNxi6r1i3vBe1tJQl+Oht2St9VIH3Eq27MZLkojA=";
+  };
+
+  patches = [
+    ./patches/extension-no-download-podman.patch # podman should be installed with nix; disable auto-installation
+    ./patches/fix-yarn-lock-deterministic.patch
+  ];
+
+  postPatch = ''
+    for file in packages/main/src/tray-animate-icon.ts extensions/podman/src/util.ts packages/main/src/plugin/certificates.ts; do
+      substituteInPlace "$file" \
+        --replace 'process.resourcesPath'          "'$out/opt/podman-desktop/resources'" \
+        --replace '(process as any).resourcesPath' "'$out/opt/podman-desktop/resources'"
+    done
+  '';
+
+  ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  nativeBuildInputs = [
+    yarn
+    fixup_yarn_lock
+    nodejs
+    makeWrapper
+    copyDesktopItems
+  ]
+  ++ lib.optionals stdenv.isDarwin [
+    desktopToDarwinBundle
+  ];
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export HOME="$TMPDIR"
+    yarn config --offline set yarn-offline-mirror "$offlineCache"
+    fixup_yarn_lock yarn.lock
+    yarn install --offline --frozen-lockfile --ignore-platform --ignore-scripts --no-progress --non-interactive
+    patchShebangs node_modules/
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    yarn --offline run build
+    yarn --offline run electron-builder --dir \
+      --config .electron-builder.config.cjs \
+      -c.electronDist=${electron}/lib/electron \
+      -c.electronVersion=${electron.version}
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p "$out/opt/podman-desktop"
+    cp -r dist/*-unpacked/{locales,resources{,.pak}} "$out/opt/podman-desktop"
```

```suggestion
    mkdir -p "$out/share/lib/podman-desktop"
    cp -r dist/*-unpacked/{locales,resources{,.pak}} "$out/share/lib/podman-desktop"
```
we usually use something like that